src/pushsource/_impl/backend/staged/staged_cloud.py

```diff
@@ -57,12 +57,14 @@ def __build_ami_push_item(self, resources, origin, image, dest):
 
         release_kwargs = {
             "product": self.__get_product_name(build_resources.get("name")),
-            "date": datetime.now(timezone.utc).strftime("%Y%m%d"),
+            "date": release_resources.get("date")
+            or datetime.now(timezone.utc).strftime("%Y%m%d"),
```

Comment:
I noticed the `datetime.now` caused problems for tests after the original change was submitted.

I'm not sure if it should be fixed just for the tests though, maybe it should rather be fixed in general by defaulting this to `None` or making it mandatory?

The problem with the tests was that loading the same data on different days will give different results.  But that seems like a bad thing in general, not just for the tests.  I don't think it's desirable that loading a staging area today might give you different data than loading it tomorrow even when there were no changes.

The default logic also does NOT match what the doc string says for this field, which is: `"Date at which this image was generated"`.  It's not valid to assume that's today when the field is missing.

If there is to be any such logic of "when the date is missing, use today", I'd feel more comfortable if that's happening in the downstream project making use of these push items.  Downstream is in a better position to decide whether applying such a default is reasonable.

Reply:
I just read @JAVGan's last comment, isn't he saying much the same?  It is not desirable that the "date" field can change depending on which day you load the staging area, as that could mean that (say) an attempted push on one day, retried the next day, could end up pushing two different image names.

Reply:
Makes sense, I removed the date generation from this.